### PR TITLE
test add_* signatures and improve docstring testing

### PR DESCRIPTION
**Pull request by kne42**
## Changes
- rename docstring test
- improve coverage & implementation of docstring test for layers and their convenience methods
- add signature test for layers and their convenience methods
- remove `__new__` method from `KeymapMixin` to stop it from overriding class signatures

### CODE REVIEW

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -366,12 +381,25 @@ def add_image(self, image, *args, **kwargs):
...
+            multichannel=multichannel,
+            colormap=colormap,
+            clim=clim,
+            clim_range=clim_range,
+            interpolation=interpolation,
+            opacity=opacity,
+            blending=blending,
+            visible=visible,
+            name=name,
+            **kwargs,
```
Comment (sofroniewn) [inline napari/components/viewer_model.py:396]: sometimes we have `**kwargs` - sometimes we don't - see `add_points`. are they just missing in `add_points`? do we want them everywhere? i think we can drop them, but curious why we'd want them

File: napari/tests/test_viewer_layer_parity.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    method_returns = *method_returns[:-1], description
+
+    assert method_returns == (
+        'layer',
+        f':class:`napari.layers.{name}`',
+        f'The newly-created {name.lower()} layer.',
+    ), f"improper 'Returns' section of '{method_name}'"
+
+
+@pytest.mark.parametrize('layer', layers, ids=lambda layer: layer.__name__)
```
Comment (sofroniewn) [inline napari/tests/test_viewer_layer_parity.py:83]: very nice

**Review by sofroniewn — APPROVED** [review]
approved - though see my minor comment on the `kwrd` ags

File: napari/components/viewer_model.py (hunk 1 above)
Comment (kne42) [reply napari/components/viewer_model.py:396]: This PR merely ensures that layers and their respective convenience methods have the same signatures and parameter docs. As to why the `Image`, `Labels`, and `Volume` layers contain unused, undocumented `**kwargs` should be addressed elsewhere.

Comment (sofroniewn) [reply napari/components/viewer_model.py:396]: makes sense

**Pull request merged**